bob-apps/bin/bobt.rs

```diff
@@ -374,13 +374,13 @@ impl Settings {
             .map_err(|e| e.to_string())
     }
 
-    fn append_request_headers(&self, b: RequestBuilder) -> RequestBuilder {
-        let mut b = b;
-        if self.username.is_some() {
-            b = b.header("username", self.username.as_ref().unwrap());
-            if self.password.is_some() {
-                b = b.header("password", self.password.as_ref().unwrap());
-            }
+    fn append_request_headers(&self, mut b: RequestBuilder) -> RequestBuilder {
+        if let (Some(username), Some(password)) = (self.username.as_ref(), self.password.as_ref()) {
+            let credentials = format!("{}:{}", username, password);
+            let credentials = base64::encode(credentials);
+            let authorization = format!("Basic {}", credentials);
+
+            b = b.header("authorization", authorization);
```

Comment:
Here is a function for basic auth: https://docs.rs/reqwest/0.11.13/reqwest/struct.RequestBuilder.html#method.basic_auth